0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1181 @@
+  "state": "rejected",
+  "timeline": {
+    "prepared_at": "2016-10-21T22:45:01.000Z",
+    "cancelled_at": "2016-10-22T21:33:34.867Z"
+  },
+  "rejection_reason": "BlacklistedSender"
+}
+```
+
+#### Errors
+
+- [NotFoundError][] - The transfer does not exist.
+- [UnprocessableEntityError][] - The request is formatted properly but contains an otherwise-unspecified semantic problem. This includes cases where the transfer has already been executed or rejected.
+- [InvalidBodyError][] - The request body was not properly-formatted JSON.
+- [TransferStateError][] - The transfer was not in the `prepared` state when the request was received. This occurs if the transfer has already been rejected or executed.
+
+
+### Get Transfer
+[Get Transfer]: #get-transfer
+
+Check the details or status of a local transfer.
+
+**Authorization:** The owners of the `debit_account` and `credit_account` MUST be able to get the transfer. A ledger MAY allow other account owners or unauthorized clients to get a transfer.
+
+#### Request Format
+
+GET the path in the `id` field of a transfer object.
+
+**Metadata URL Name:** `transfer`. Suggested generic path: `/transfers/{client_id}`
+
+#### Response Format
+
+A successful result uses the HTTP response code **200 OK** and contains a JSON body with the requested [Transfer resource][].
+
+#### Example
+
+Request:
+
+```
+GET /ledger/transfers/63a61fa9-fca2-4779-9d50-49dc691b8fbf
+```
+
+Response:
+
+```json
+HTTP/1.1 200 OK
+
+{
+  "ledger": "https://red.ilpdemo.org/ledger",
+  "client_id": "63a61fa9-fca2-4779-9d50-49dc691b8fbf",
+  "debit_account": "https://red.ilpdemo.org/ledger/accounts/alice",
+  "credit_account": "https://red.ilpdemo.org/ledger/accounts/bob",
+  "amount": "199.99",
+  "execution_condition": "cc:0:3:dB-8fb14MdO75Brp_Pvh4d7ganckilrRl13RS_UmrXA:66",
+  "expires_at": "2018-01-01T00:00:00.000Z",
+  "memo": {
+    "ilp_header": {
+      "version": 1,
+      "destinationAddress": "ilpdemo.blue.bob",
+      "destinationAmount": "199.0"
+    }
+  },
+  "note_to_self": {
+    "foo": "bar"
+  },
+  "id": "https://red.ilpdemo.org/ledger/transfers/63a61fa9-fca2-4779-9d50-49dc691b8fbf",
+  "fulfillment": "https://red.ilpdemo.org/ledger/transfers/63a61fa9-fca2-4779-9d50-49dc691b8fbf/fulfillment",
+  "state": "rejected",
+  "timeline": {
+    "prepared_at": "2016-10-21T22:45:01.000Z",
+    "cancelled_at": "2016-10-22T21:33:34.867Z"
+  },
+  "rejection_reason": "BlacklistedSender"
+}
+```
+
+#### Errors
+
+- [NotFoundError][]
+- [InvalidUriParameterError][]
+
+
+### Get Transfer Fulfillment
+[Get Transfer Fulfillment]: #get-transfer-fulfillment
+
+Retrieve the fulfillment for a transfer that has been executed or canceled. This is separate from the [Transfer resource][] because it can be very large.
+
+**Authorization:** The ledger MUST allow owners of the `credit_account` and `debit_account` to get the transfer's fulfillment. A ledger MAY allow other account owners or unauthorized users to get a transfer's fulfillment.
```

Comment:
Might be helpful to include a note about why you would or would not allow other users to get a transfer's fulfillment. If you can't get another user's transfer you probably shouldn't be able to get their fulfillments (which would indicate that a transfer has happened). Not sure how this would work with the dependent transfers idea...